include/bluetooth/mesh/main.h

```diff
@@ -432,7 +432,8 @@ int bt_mesh_lpn_poll(void);
  *
  *  @param cb Function to call when the Friendship status changes.
  */
-void bt_mesh_lpn_set_cb(void (*cb)(uint16_t friend_addr, bool established));
+void bt_mesh_lpn_set_cb(void (*cb)(uint16_t friend_addr, uint16_t queue_size,
+	                      uint8_t recv_win, bool established));
```

Comment:
Related to @MarcianoPreciado's comments on the LPN API in #27842, could you instead make this a structure with a callback, so we can add it to a linked list in the LPN code? That way, we can add more listeners for the LPN events, and we don't have to break this API again on the next release if we want to add more callbacks for events such as polls.
e.g. something like this?
```C
struct bt_mesh_lpn_cb {
    void (*connected)(uint16_t friend_addr, uint16_t queue_size, uint8_t recv_window);
    void (*disconnected)(void);
    void (*polled)(void);
    sys_snode_t node;
};
```